test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs

```diff
@@ -854,6 +854,42 @@ public async Task MsbuildRestore_WithMissingProjectReferences_HandlesProjectRefe
             }
         }
 
+        [PlatformTheory(Platform.Windows)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void MsbuildRestore_WithUnsupportedProjects_Warns(bool restoreUseStaticGraphEvaluation)
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var net461 = NuGetFramework.Parse("net472");
+                var project = new SimpleTestProjectContext("b", ProjectStyle.PackageReference, pathContext.SolutionRoot);
+
+                solution.Projects.Add(project);
+                solution.Create(pathContext.SolutionRoot);
+
+                File.WriteAllText(
+                   project.ProjectPath,
+                   @"<Project />");
+
+                var result = _msbuildFixture.RunMsBuild(pathContext.WorkingDirectory, $"/t:restore /p:RestoreUseStaticGraphEvaluation={restoreUseStaticGraphEvaluation} {solution.SolutionPath}", ignoreExitCode: true);
+
+                // Assert
+                result.ExitCode.Should().Be(0, result.AllOutput);
+                if (restoreUseStaticGraphEvaluation)
+                {
+                    result.AllOutput.Should().Contain($"Skipping restore for project '{project.ProjectPath}'. The project file may be invalid or missing targets required for restore.");
```

Comment:
Would it make sense for this code to call `MSBuildRestoreUtility.GetMessageForUnsupportedProject()` so we don't have unlocalized strings in new tests?